app/components/curriculum-inventory/sequence-block-overview.js

```diff
@@ -220,15 +220,15 @@ export default class CurriculumInventorySequenceBlockOverviewComponent extends C
     this.minimum = this.args.sequenceBlock.minimum;
   }
 
-  @dropTask
-  *saveCourse() {
-    const oldCourse = yield this.args.sequenceBlock.course;
+  @action
+  async saveCourse() {
+    const oldCourse = await this.args.sequenceBlock.course;
     if (oldCourse !== this.course) {
       this.args.sequenceBlock.set('sessions', []);
       this.args.sequenceBlock.set('excludedSessions', []);
     }
     this.args.sequenceBlock.set('course', this.course);
```

Comment:
This is where we get into trouble with an action, as this `set` is called after the `await` it is possible for this component to get torn down and `this` to be invalid during the async work. To be safe this approach should protect against that with `if (!this.isDestroying) {`.

Reply:
:heavy_check_mark: